test/node-test/client-dispatch.js

```diff
@@ -74,33 +76,33 @@ test('dispatch invalid opts', (t) => {
     bodyTimeout: 'asd'
   }, {
     onError (err) {
-      t.type(err, errors.InvalidArgumentError)
-      t.equal(err.message, 'invalid bodyTimeout')
+      p.ok(err instanceof errors.InvalidArgumentError)
+      p.strictEqual(err.message, 'invalid bodyTimeout')
     }
   })
 
   client.dispatch(null, {
     onError (err) {
-      t.type(err, errors.InvalidArgumentError)
-      t.equal(err.message, 'opts must be an object.')
+      p.ok(err instanceof errors.InvalidArgumentError)
+      p.strictEqual(err.message, 'opts must be an object.')
     }
   })
 })
 
-test('basic dispatch get', (t) => {
-  t.plan(11)
+test('basic dispatch get', async (t) => {
+  const p = tspl(t, { plan: 11 })
 
   const server = http.createServer((req, res) => {
-    t.equal('/', req.url)
-    t.equal('GET', req.method)
-    t.equal(`localhost:${server.address().port}`, req.headers.host)
-    t.equal(undefined, req.headers.foo)
-    t.equal('bar', req.headers.bar)
-    t.equal('', req.headers.baz)
-    t.equal(undefined, req.headers['content-length'])
+    p.strictEqual('/', req.url)
+    p.strictEqual('GET', req.method)
+    p.strictEqual(`localhost:${server.address().port}`, req.headers.host)
+    p.strictEqual(undefined, req.headers.foo)
+    p.strictEqual('bar', req.headers.bar)
+    p.strictEqual('', req.headers.baz)
+    p.strictEqual(undefined, req.headers['content-length'])
     res.end('hello')
   })
-  t.teardown(server.close.bind(server))
+  t.after(server.close.bind(server))
```

Comment:
Unfortunately this does not work the same. server.close is a callback API, while t.after expects returning a Promise. Can you write a tiny adapter to match the two?

Reply:
What do you think about the interface below?

```js
// definition
function closeServerAsPromise(server) {
  return () => util.promisify(server.close.bind(server))()
}

// usage
t.after(closeServerAsPromise(server))
```

Reply:
That works

Reply:
I've created new PR that adds `closeServerAsPromise` helper for existing tests. https://github.com/nodejs/undici/pull/2604